apps/nearbyflight/nearby_flight.star

```diff
@@ -0,0 +1,196 @@
+                                                    pad = (0, 3, 0, 0),
+                                                ),
+                                                render.Text("", color = "#0F0", font = "tom-thumb"),
+                                                render.Text("OPEN", color = "#666", font = "tom-thumb"),
+                                                render.Text("FLIGHT", color = "#666", font = "tom-thumb"),
+                                            ],
+                                        ),
+                                        render.Column(
+                                            expanded = True,
+                                            main_align = "start",
+                                            cross_align = "center",
+                                            children = [
+                                                render.Padding(
+                                                    child = render.Image(src = qr_code_for_flight(nearest)),
+                                                    pad = (5, 1, 0, 0),
+                                                ),
+                                            ],
+                                        ),
+                                    ],
+                                ),
+                            ],
+                        ),
+                    ],
+                ),
+            ],
+        ),
+    )
+
+def qr_code_for_flight(flight):
+    url = "https://app.airframes.io/flights/%s" % int(flight["id"])
+    print("Generating QR code for %s" % url)
+    code = qrcode.generate(
+        url = url,
+        size = "large",
+        color = "#FFF",
+        background = "#000",
+    )
+    return code
+
+def get_nearby_flights(api_key, lat, lon, radius):
+    cache_key = "nearby_%s_%s_%s" % (lat, lon, radius)
+    nearby_cached = cache.get(cache_key)
+    if nearby_cached != None:
+        print("Hit! Displaying cached data.")
+        return json.decode(nearby_cached)
+
+    print("Miss! Calling Airframes API.")
+
+    rep = http.get(AIRFRAMES_NEARBY_URL, params = {
+        "lat": str(lat),
+        "lon": str(lon),
+        "radius": radius,
+        "limit": AIRFRAMES_NEARBY_LIMIT,
+        "sort": AIRFRAMES_NEARBY_SORT,
+        "sort_order": AIRFRAMES_NEARBY_SORT_ORDER,
+    }, headers = {
+        "X-API-KEY": api_key or "none",
+    })
+    print(rep.url)
+    if rep.status_code != 200:
+        fail("Airframes API request failed with status %d" % rep.status_code)
+
+    nearby = rep.json()
+    cache.set(cache_key, json.encode(nearby), ttl_seconds = 60)
+    return nearby
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
+                id = "api_key",
+                name = "Airframes API Key",
+                desc = "Your Airframes API key. You can find this in your Airframes account settings.",
```

Comment:
Note that these field description are currently not visible to users in the Tidbyt iOS/Android app. You'll need to add the information to the app description in the manifest file if you want people to see it.

Reply:
Definitely, makes sense.